main/src/library/List.flix

```diff
@@ -646,7 +646,7 @@ namespace List {
     ///
     /// That is, the result is of the form: `f(x1, ...f(xn-1, f(xn, s))...)`.
     ///
-    pub def foldRight(f: (a, b) -> b & ef, s: b, l: List[a]): b & ef =
+    pub def foldRightOld(f: (a, b) -> b & ef, s: b, l: List[a]): b & ef =
```

Comment:
I kept the old `foldRight`, since it cannot be deleted until `Foldable` is updated? Is this okay?
My thought was to keep track of all the missing pieces in #3226 and fix this when the other namespaces are updated and ready.